src/main/java/uk/gov/homeoffice/digital/sas/timecard/listeners/TimeEntryKafkaEntityListener.java

```diff
@@ -1,57 +1,43 @@
 package uk.gov.homeoffice.digital.sas.timecard.listeners;
 
-import jakarta.persistence.PostPersist;
-import jakarta.persistence.PostRemove;
-import jakarta.persistence.PostUpdate;
-import java.util.Arrays;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreRemove;
+import jakarta.persistence.PreUpdate;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.core.env.Environment;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
 import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
 
 @Component
 public class TimeEntryKafkaEntityListener extends KafkaEntityListener<TimeEntry> {
 
-  private Environment environment;
+  @Value("${spring.kafka.template.default-topic}")
+  private String topicName;
 
   @Override
   public String resolveMessageKey(TimeEntry timeEntry) {
-    return timeEntry.getOwnerId().toString();
+    return String.valueOf(timeEntry.getTenantId() + ":" + timeEntry.getOwnerId());
```

Comment:
nit: don't think you need to do String.valueOf here. - Implicit concatenation

Reply:
Actioned